### PR TITLE
chore: Nix setup

### PR DESCRIPTION
**Pull request by lukelooker**
Adds support for Nix

### CODE REVIEW

**Review by ElliotTheHuman — APPROVED** [review]
I'm not too saavy in Nix, but I don't see anything egregious to block this. Happy to 👍

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,6 @@ yarn-debug.log*
 yarn-error.log*
 /lerna-debug.log
 
+# nix
+cache-*
+drv
```
Comment (mdodgelooker) [inline .gitignore:50]: I couldn't see these folders show up for me after starting a `nix-shell` but I assume they show up at some point?

**Review by mdodgelooker — COMMENTED** [review]
LGTM! (I didn't see the newly ignored folders though)

**Review by mdodgelooker — APPROVED** [review]
Sorry, meant to approve this in spite of not seeing those folders

File: .gitignore (hunk 1 above)
Comment (lukelooker) [reply .gitignore:50]: They did for me at least :)

**Pull request merged**